File: pfactor.js
const io = require('console-read-write');

function factor(number){
  const factors = [];
  let i=2;
  while(number>=2){ //&& i < Math.floor(number/2)){
    if(number % i === 0){
      factors.push(i);
      number /= i;
      i = 2;
      continue;
    }
    i++;
  }
  return factors;
}

function print_factors(num){
  const factors = factor(num);
  console.log("The prime factors of " + num + " are: " + factors.toString())
}

async function main(){
  while(true){
	  console.log("Enter a number to factor:");
	  let inp = await io.read();
	  if(inp==="exit"){
		  console.log("Goodbye.");
		  break;
	  }
	  print_factors(inp);
	  console.log("Enter \"exit\" to exit.\n");
  }
}

main();

File: pfactor.cpp
#include <iostream>
#include <limits>

using namespace std;

#define MAX_INT 2147483646


int factors[256];

int* factor(int number){
  int c=0;
  int i=2;
  while(number>=2){
    if(number % i == 0){
      factors[c] = i;
      c++;
      number /= i;
      i = 2;
      continue;
    }
    i++;
  }
  return factors;
}

void print_array(int* arr){
  int j=0;
  while(arr[j] > 0){
    cout << arr[j];
    if(arr[++j] > 0) cout << ", ";
  }
  cout << endl;
  return;
}

int main(){
  int n;
  cout << "Enter a number between 2 and 2^31 to prime factorize:\n";
  do{
	cin.clear();
	cin >> n;		
	if(cin.fail()){
		cout << "Please enter a number.\n";
		cin.clear();
		cin.ignore(std::numeric_limits<streamsize>::max(),'\n');
	}
  }while(n > MAX_INT || n < 2);
  cout << "The prime factors of " << n << " are:\n";
  print_array(factor(n));
  return 0;
}

